monailabel/transform/post.py

```diff
@@ -37,23 +37,52 @@
 
 
 # TODO:: Move to MONAI ??
-
-
 class LargestCCd(MapTransform):
     def __init__(self, keys: KeysCollection, has_channel: bool = True):
+        """
+        MapTransform that applies the get_largest_cc function to label images in a data dictionary.
+
+        Args:
+            keys: keys of the data dictionary to be transformed.
+            has_channel: whether the data has channel dimension or not.
+        """
         super().__init__(keys)
         self.has_channel = has_channel
 
     @staticmethod
     def get_largest_cc(label):
-        largest_cc = np.zeros(shape=label.shape, dtype=label.dtype)
-        for i, item in enumerate(label):
-            item = measure.label(item, connectivity=1)
```

Comment:
i believe the only change u need is

```python
item = ndimage.label(item, connectivity=1)
if item.max() != 0:
  largest_cc[i, ...] = item == ndimage.mode(item, axis=None)[0][0]
```

which is using `ndimage.label` and `ndimage.mode`

if see largest_cc already has the shape of original labels and all initialized to zeros..
so you only pick one label and try to update the corresponding index if the label is valid (max != 0)